SourceKitStressTester/Sources/StressTester/StressTester.swift

```diff
@@ -45,7 +39,64 @@ public struct StressTester {
     }
   }
 
-  func computeStartStateAndActions(from tree: SourceFileSyntax) -> (state: SourceState, actions: [Action]) {
+  public func run(for file: URL, swiftc: String, compilerArgs: [String]) throws {
+    let compilerArgs = compilerArgs.flatMap {
+      DriverFileList(at: $0)?.paths ?? [$0]
+    }
+    var document = SourceKitDocument(file,
+                                     swiftc: swiftc,
+                                     args: compilerArgs,
+                                     tempDir: options.tempDir,
+                                     connection: connection,
+                                     containsErrors: true)
+
+    // compute the actions for the entire tree
+    let (tree, _) = try document.open(rewriteMode: options.rewriteMode)
+    let (actions, replacements) = computeActions(from: tree)
+
+    if let dryRunAction = options.dryRun {
+      try dryRunAction(actions)
+      return
+    }
+
+    if !replacements.isEmpty {
+      // Update initial state
+      _ = try document.update() { sourceState in
+        for action in replacements {
+          if case .replaceText(let offset, let length, let text) = action {
```

Comment:
You can inline the case part into the for, i.e. `for case .replaceText(let offset, let length, let text) in replacements`.
This also makes the filtering you're doing when computing the replacements redundant, but I'm not sure what you'd call them without that. "priorActions"?

Reply:
priorActions is a good name